synth.py

```diff
@@ -21,21 +21,27 @@
 
 logging.basicConfig(level=logging.DEBUG)
 
-gapic = gcp.GAPICGenerator()
+gapic = gcp.GAPICMicrogenerator()
 common_templates = gcp.CommonTemplates()
 
 versions = ['v1', 'v1beta1']
 
 for version in versions:
-    library = gapic.node_library('texttospeech', version)
+    library = gapic.typescript_library(
+        'texttospeech', 
+        generator_args={
+            "grpc-service-config": f"google/cloud/texttospeech/{version}/texttospeech_grpc_service_config.json"
+            },
+        proto_path=f'/google/cloud/texttospeech/{version}',
+        version=version)
 
     # skip index, protos, package.json, and README.md
     s.copy(
         library,
-        excludes=['package.json', 'README.md', 'src/index.js'],
+        excludes=['package.json', 'README.md', 'src/index.ts'],
```

Comment:
This is a feature request @alexander-fenster, but I really wish we could generate this index file

Reply:
This is a feature request for `synthtool` :)  It's by design out of scope of the generator.

(but I agree it should be automated)

Reply:
I mean - I totally hear where you're coming from.  At the same time, we're building this generator so we don't have to go mucking about in Java when we want to make a change. Sticking per-language logic in synthtool just means we have to go mucking about in Python instead.

Realistically, today we're managing ~40 of these hand written files to make up for not having generation for it.  We have to add it.  It can be in synthtool or the generator, but I would personally rather have us write it in JavaScript so we control our own destiny a bit more.

Reply:
Okay, I will think how to take care of this. It does not fit well into the generator API (described in [AIP-4290](https://aip.dev/4290)) but it surely can be done on some level.